docs/release.md

```diff
@@ -0,0 +1,53 @@
+# Release process of the RTE Yocto BSP
```

Comment:
@lukasz-laguna Maybe we should somehow merge this description with the: https://github.com/3mdeb/meta-rte#development-workflow